### PR TITLE
Update maintainer guide for releasing

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

None, as it's a documentation fix.

> Is there anything in the PR that needs further explanation?

This pull request just fixes small mistakes in the guide.

### CODE REVIEW

File: docs/maintainer-guide/releases.md [hunk 1]
```diff
@@ -1,29 +1,29 @@
 # Performing releases
 
 1. Create a [new issue](https://github.com/stylelint/stylelint/issues/new) announcing the planned release, e.g. `Release 8.11.1` and include the [template checklist](#new-release-issue-template).
-2. Locally test `main` in the `stylelint-config-*` shareable config repositories. Install current `main` branch (`npm install stylelint/stylelint#main`) and run tests.
-3. Locally test `main` in the [stylelint/stylelint.io](https://github.com/stylelint/stylelint.io) repository.
-4. Locally test `main` in the [stylelint/stylelint-demo](https://github.com/stylelint/stylelint-demo) repository.
+2. Locally test `main` in the `stylelint-config-*` shareable config repositories. Install the current `main` branch (`npm install stylelint/stylelint#main --save-dev`) and run tests.
```
Comment (ybiquitous) [inline docs/maintainer-guide/releases.md:4]: [note] The `--save-dev` flag is required. Without it, the command fails:

```console
$ npm install stylelint/stylelint#main
npm ERR! Invalid comparator: github:stylelint/stylelint#main
```

File: docs/maintainer-guide/releases.md [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
+3. Locally test `main` in the [stylelint/stylelint-demo](https://github.com/stylelint/stylelint-demo) repository.
+4. Locally test `main` in the [stylelint/stylelint.io](https://github.com/stylelint/stylelint.io) repository.
```
Comment (ybiquitous) [inline docs/maintainer-guide/releases.md:6]: [note] The updates are done in the order of `stylelint-demo` and `stylelint.io`.

File: docs/maintainer-guide/releases.md [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
    4. Push these changes.
-   5. Confirm the changes are correct at [https://github.com/stylelint/stylelint](https://github.com/stylelint/stylelint).
+   5. Confirm the changes are correct at [github.com/stylelint/stylelint](https://github.com/stylelint/stylelint).
    6. Run `npm run release`.
    7. Select the version that matches the one from the changelog.
    8. Copy and paste the changelog entries for the published version from [changelog](../../CHANGELOG.md) when the GitHub release page opens.
-   9. Confirm the publishing of the package to [https://www.npmjs.com/package/stylelint](https://www.npmjs.com/package/stylelint).
-   10. Confirm the creation of the release at [https://github.com/stylelint/stylelint/releases](https://github.com/stylelint/stylelint/releases).
+   9. Confirm the publishing of the package to [www.npmjs.com/package/stylelint](https://www.npmjs.com/package/stylelint).
+   10. Confirm the creation of the release at [github.com/stylelint/stylelint/releases](https://github.com/stylelint/stylelint/releases).
```
Comment (ybiquitous) [inline docs/maintainer-guide/releases.md:17]: [note] For consistency and simplicity, I've omitted the `https://` protocol from the links.

**Review by jeddy3 — APPROVED** [review]
All LGTM, thanks!

**Pull request merged**